src/t_stream.c

```diff
@@ -986,7 +986,7 @@ static int streamParseAddOrTrimArgsOrReply(client *c, streamAddTrimArgs *args, i
             if (args->approx_trim) {
                 /* In order to prevent from trimming to do too much work and cause
                  * latency spikes we limit the amount of work it can do */
-                args->limit = 100 * server.stream_node_max_entries;
+                args->limit = 100 * server.stream_node_max_entries; /* Maximum 100 rax nodes. */
```

Comment:
i think the right solution to this discussion, is to just clamp the result of this multiplication to some range. e.g. 1 .. 32k